azure-toolkit-libs/azure-toolkit-appservice-lib/src/main/java/com/microsoft/azure/toolkits/appservice/service/impl/WebAppDeploymentSlot.java

```diff
@@ -34,81 +39,163 @@ public WebAppDeploymentSlot(WebAppDeploymentSlotEntity deploymentSlot, AzureAppS
+    public class WebAppDeploymentSlotCreator implements IWebAppDeploymentSlotCreator {
+        public static final String CONFIGURATION_SOURCE_NEW = "new";
+        public static final String CONFIGURATION_SOURCE_PARENT = "parent";
+        private static final String CONFIGURATION_SOURCE_DOES_NOT_EXISTS = "Target slot configuration source does not exists in current web app";
+
+        private String name;
+        private String configurationSource = CONFIGURATION_SOURCE_PARENT;
+        private Optional<Map<String, String>> appSettings = null;
+
+        @Override
+        public IWebAppDeploymentSlotCreator withName(String name) {
+            this.name = name;
+            return this;
+        }
+
+        @Override
+        public IWebAppDeploymentSlotCreator withAppSettings(Map<String, String> appSettings) {
+            this.appSettings = Optional.ofNullable(appSettings);
+            return this;
+        }
+
+        @Override
+        public IWebAppDeploymentSlotCreator withConfigurationSource(String configurationSource) {
+            this.configurationSource = configurationSource;
+            return this;
+        }
+
+        @Override
+        public WebAppDeploymentSlot commit() {
+            final WebAppDeploymentSlotEntity entity = WebAppDeploymentSlot.this.entity();
+            final WebApp webApp = azureClient.webApps().getByResourceGroup(entity.getResourceGroup(), entity.getWebappName());
+            final DeploymentSlot.DefinitionStages.Blank blank = webApp.deploymentSlots().define(getName());
+            final DeploymentSlot.DefinitionStages.WithCreate withCreate;
+            // Using configuration from parent by default
+            final String source = StringUtils.isEmpty(configurationSource) ? CONFIGURATION_SOURCE_PARENT : StringUtils.lowerCase(configurationSource);
+            switch (source) {
+                case CONFIGURATION_SOURCE_NEW:
+                    withCreate = blank.withBrandNewConfiguration();
+                    break;
+                case CONFIGURATION_SOURCE_PARENT:
+                    withCreate = blank.withConfigurationFromParent();
+                    break;
+                default:
+                    final DeploymentSlot deploymentSlot = deploymentSlotClient.parent().deploymentSlots().getByName(configurationSource);
+                    if (deploymentSlot == null) {
+                        throw new RuntimeException(CONFIGURATION_SOURCE_DOES_NOT_EXISTS);
+                    }
+                    withCreate = blank.withConfigurationFromDeploymentSlot(deploymentSlot);
+                    break;
+            }
+            if (appSettings != null && appSettings.isPresent()) {
+                withCreate.withAppSettings(appSettings.get());
+            }
+            WebAppDeploymentSlot.this.deploymentSlotClient = withCreate.create();
+            WebAppDeploymentSlot.this.slotEntity = AppServiceUtils.getWebAppDeploymentSlotEntity(WebAppDeploymentSlot.this.deploymentSlotClient);
```

Comment:
convertToWebAppDeploymentSlotEntity?

Reply:
now convert to SDK models is all named `toXXX()`, convert to library model is all named `fromXXX()`, do we need to update the naming? As we discussed before, we could indicate the method from parameter type

Reply:
yes, toxx or fromxxx is ok, bug getXXX is not ok, it makes reader thinking that the entity is retrieved through api call to remote

Reply:
change to `fromWebAppDeploymentSlot`